### DISCUSSION TITLE
follow an exact playlist sequence

### DISCUSSION THREAD

**Discussion by Marcioghiraldelli** [AzuraCast/AzuraCast#7012] [Show and tell]
I've been researching a way to force Azuracast to follow an exact sequence of what I want.
example

playlist 1
playlist 2
Playlist 3
playlist 2
playlist 1

and so on...
I searched in several places and I wasn't able to find something that could suit me...
Can anyone help me with a liquidsoap script that can be used in Azuracast where it would be possible to do this?

**Comment by gAlleb** [comment]
Hi! Only this comes to mind:

1. You can make this playlists `advanced` (In "Basic Info" settings of Playlist you choose "Advanced") 
2. And then --> `Edit Liquidsoap Configuration` 3rd box (below `# Apply amplification metadata (if supplied) radio = amplify(override="liq_amplify", 1., radio)`):

```
radio = rotate(weights=[1,1,1,1], [playlist_1, playlist_2, playlist_3,playlist_2])
```

**Reply by Marcioghiraldelli** [reply]
I understand, but if I want each playlist to play differently, is it possible?
example: playlist 1 plays all the audios within it, playlist 2 only one audio and playlist 3 plays the audios within it randomly.

**Reply by gAlleb** [reply]
You are not specific thus It's difficult to understand what you want.

> playlist 1 plays all the audios within it

For this you can merge tracks within it either in playlist settings or by hand:
```
playlist_1 = playlist(id="playlist_1",mime_type="audio/x-mpegurl",mode="randomize",reload_mode="watch","file_path")
playlist_1 = merge_tracks(id="playlist_1", playlist_1)
```
Or just specify the number of tracks in `rotate` which equals the number of tracks in a playlist. If it has 130 tracks - `radio = rotate(weights=[130,1,1], [playlist_1, playlist_2, playlist_3])`

> playlist 2 only one audio and playlist 3 plays the audios within it randomly. 

You want playlist_2 to play files sequentially and 3rd one randomly? Then you can add `mode="normal"` and `loop=true` (if you want to reload a playlist each round - `reload=1, reload_mode="rounds"`)

```
playlist_2 = playlist(id="playlist_2",mime_type="audio/x-mpegurl",mode="normal",reload_mode="watch", loop=true,"file_path")
playlist_3 = playlist(id="playlist_3",mime_type="audio/x-mpegurl",mode="randomize",reload_mode="watch","file_path")
```

Result:
```
playlist_1 = playlist(id="playlist_1",mime_type="audio/x-mpegurl",mode="randomize",reload_mode="watch","file_path")
playlist_1 = merge_tracks(id="playlist_1", playlist_1)
playlist_2 = playlist(id="playlist_2",mime_type="audio/x-mpegurl",mode="normal", loop=true, reload_mode="watch", "file_path")
playlist_3 = playlist(id="playlist_3",mime_type="audio/x-mpegurl",mode="randomize",reload_mode="watch","file_path")
radio = rotate(weights=[1,1,1], [playlist_1, playlist_2, playlist_3])
```
playlist_1 will play all files, then we would have one file from playlist_2 (files play in order), then  random file from playlist_3. Then again playlist_1 will play all files. (Can't imagine use case for it. But that's how you wanted:)) Not completely sure if it would work the desired way so no 100% guarantee. 

But with all this different settings you can manipulate how your playlists behave (sequentially or randomly, merged or not, how many tracks per playlist to play).
